makelib/kubebuilder.mk

```diff
@@ -63,4 +100,13 @@ $(KUBEBUILDER):
 	@rm -fr $(TOOLS_HOST_DIR)/tmp
 	@$(OK) installing kubebuilder $(KUBEBUILDER_VERSION)
 
-
+$(CONTROLLERGEN):
+	@$(INFO) installing controller-gen @$(CONTROLLER_GEN_VERSION)
+	$(eval PKG_PATH := sigs.k8s.io/controller-tools/cmd/controller-gen)
+ifeq ($(GO111MODULE),on)
+	@$(MAKE) go.get GOPKG=$(PKG_PATH)@$(CONTROLLER_GEN_VERSION)
+else
+	@cd vendor/$(PKG_PATH);GOBIN=$(TOOLS_HOST_DIR) go install 
```

Comment:
this will install things in the user's home directory, no? Also I don't get why there is a conditional on `GO111MODULE`

Reply:
Currently (w/o `go modules`), the `controller-gen` package is tracked by `dep`, and resides in `$(PROJECT_FOLDER)/vendor`. The `@cd vendor/$(PKG_PATH);GOBIN=$(TOOLS_HOST_DIR) go install` builds and installs the binary in `$(TOOLS_HOST_DIR)`

This tool was being used before by calling the `main.go` file directly, which I thought I should create its binary to be compatible with other tools we use. See https://github.com/crossplaneio/stack-aws/pull/20/files#diff-795fc8c20b8a15e3595e13f3119822e4L18 for the proposed change in stack-aws repository.

Reply:
this logic seems specific to the crossplane project then? I'm not seeing what his makefile is making assumptions about the `vendor/` dir of crossplane.

Reply:
changed to install the package regardless of consumer repositories `vendor/` dir